sdk/core/core/src/az_uri.c

```diff
@@ -7,25 +7,22 @@
 
 #include <assert.h>
 #include <stdbool.h>
+#include <ctype.h>
 
 #include <_az_cfg.h>
 
 AZ_NODISCARD AZ_INLINE bool should_encode(uint8_t const c) {
   switch (c) {
-    default:
-      return !(('0' <= c && c <= '9') || ('A' <= c && c <= 'Z') || ('a' <= c && c <= 'z'));
     case '-':
     case '_':
     case '.':
     case '~':
       return false;
+    default:
+      return !(('0' <= c && c <= '9') || ('A' <= c && c <= 'Z') || ('a' <= c && c <= 'z'));
```

Comment:
```suggestion
      return !isalnum(c);
```

Reply:
Please, make sure `isalnum` is really only Latin letters.